### ISSUE TITLE
Compile Error on running Demo

### ISSUE THREAD

**Issue by hcheng826** [ZenGo-X/multi-party-bls#20]
Hi team,

I was trying to run the demo with the following command from README.
```
cargo run --example cli -- mediator-server run
```

But the error below happens.
```
Compiling bls v0.1.4 (.../multi-party-bls)
error: linking with `cc` failed: exit status: 1

= note: "cc" "-arch" "arm64" "/var/folders/_d/f5xtszqd1wv1hmyww1cdrmx40000gn/T/rustcFuU4oo/symbols.o" ... and lots of files

ld: library not found for -lgmp
          clang: error: linker command failed with exit code 1 (use -v to see invocation)

error: could not compile `bls` due to previous error
```
I am using a M1 Macbook and this is my cargo version: `cargo 1.61.0 (a028ae42f 2022-04-29)`

Is there anyone else have the same problem or I missed any pre-configuration?
Appreciate it!

**Comment by hcheng826** [comment]
https://github.com/rust-lang/rust/issues/25289
should be a general issue for rust.

**Comment by hcheng826** [comment]
Tried a couple of approaches found but none of them actually works for me. Decide to reopen it to see if anyone has the same issue.

Sharing some links I found:
https://github.com/madMAx43v3r/chia-plotter/issues/519
https://substrate.stackexchange.com/questions/6058/error-linking-with-cc-failed-exit-status-1-mac-m1
https://stackoverflow.com/questions/28124221/error-linking-with-cc-failed-exit-code-1

Is there any prerequisite package or library? Before I can just run
```
cargo run --example cli -- mediator-server run
```
